### PR TITLE
Add `--check` flag for checking if the local repository matches

### PR DESCRIPTION
**Pull request by Mrtenz**
## Description

This pull request adds a new `--check` flag, which can be used for checking if the local repository matches the module template, without making any modifications.

## Changes

1. Refactor CLI to use Yargs.
2. Add `--check` option, and handle it in all handlers.
3. Update README.md to include information about the new `--check` option.

### CODE REVIEW

File: src/processing/yarnrc.ts [hunk 1]
```diff
@@ -2,17 +2,74 @@ import { assert, isPlainObject } from '@metamask/utils';
...
+  const legacyYarnRcExists = await pathExists(LEGACY_YARN_PATH);
+  if (!legacyYarnRcExists) {
+    return;
+  }
+
+  if (check) {
+    warn(
+      spinner,
+      'Legacy .yarnrc file exists. It will not be deleted because the --check flag is enabled.',
+    );
```
Comment (FrederikBolding) [inline src/processing/yarnrc.ts:35]: Missing return, meaning this message is a lie 😅

File: src/processing/package-json.ts [hunk 2]
```diff
@@ -80,39 +90,56 @@ export async function processPackageJson(spinner: Ora) {
       }
 
       if (!currentPackageJson.scripts[name]) {
-        log(spinner, chalk.dim(`Adding script "${chalk.reset(name)}".`));
+        if (check) {
+          warn(
+            spinner,
+            `Local "${name}" script is missing. It should be "${script}".`,
+          );
```
Comment (FrederikBolding) [inline src/processing/package-json.ts:97]: Needs a `continue`?

File: src/processing/package-json.ts [hunk 3]
```diff
@@ -60,14 +62,22 @@ export async function processPackageJson(spinner: Ora) {
...
-            `Updating dependency "${chalk.bold(name)}" to "${chalk.bold(
-              version,
-            )}"`,
-          ),
-        );
+        if (check) {
+          warn(
+            spinner,
+            `Local "${name}" dependency is out of date. It should be "${version}".`,
+          );
```
Comment (FrederikBolding) [inline src/processing/package-json.ts:69]: Needs a `continue`?

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: src/processing/yarnrc.ts (hunk 1 above)
Comment (Mrtenz) [reply src/processing/yarnrc.ts:35]: Oops. 😅

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**